src/ert/ensemble_evaluator/evaluator.py

```diff
@@ -140,13 +141,15 @@ def set_event_handler(event_types: Set[Type[Event]], func: Any) -> None:
                 and asyncio.get_running_loop().time() - start_time
                 < self._batching_interval
             ):
+                self._complete_batch.clear()
                 try:
                     event = await asyncio.wait_for(self._events.get(), timeout=0.1)
                     function = event_handler[type(event)]
                     batch.append((function, event))
                     self._events.task_done()
                 except asyncio.TimeoutError:
                     continue
+            self._complete_batch.set()
```

Comment:
Would it be over engineered if we made everything inside the loop a separate method, and had the `self._complete_batch.set/clear` in a decorator on that new method?

Reply:
I think it would be over engineered - yes! If there would be more utilization of such a decorator then I would say yes.